table.go

```diff
@@ -100,3 +100,52 @@ func (db Database) ReadTable(tableName string) (*ResultSet, error) {
 
 	return &ResultSet{Keys: keys, Rows: rows}, nil
 }
+
+func (db Database) Insert(tableName string, values map[string]interface{}) error {
+	tableEntry, err := db.findTableEntry(tableName)
+	if err != nil {
+		return err
+	}
+	tableHeaderBlock, err := db.readHeaderBlock(tableEntry.HeaderBlock)
+	if err != nil {
+		return err
+	}
+
+	block, err := db.readBlock(tableEntry.HeaderBlock)
+	if err != nil {
+		return err
+	}
+
+	recordSize, _ := tableHeaderBlock.recordReaders()
+	tableColumns := tableHeaderBlock.TableColumns()
+	rows := []Row{}
+
+	for recordBlockAddr := tableHeaderBlock.FirstRecordBlock; recordBlockAddr != nullBlockAddr; {
+		recordBlock, err := db.readRecordBlock(tableHeaderBlock.FirstRecordBlock)
+		if err != nil {
+			return err
+		}
+
+		for _, record := range recordBlock.Data.split(recordSize) {
+			if record.isFree() {
+				continue
+			}
+
+			row := Row{}
```

Comment:
En lugar de llenar el arreglo de Rows con ese for loop deberías agregar lo siguiente (en pseudocodigo):

```c
if(recordBlock.HasFreeSpace()) {
    recordBlock.insertRecord(values)
}
```

Además falta validar el caso en que todos los recordBlocks estén llenos, cuando eso sucede es necesario crear un nuevo recordBlock, enlazarlo con el último recordBlock e insertar el record en ese nuevo bloque.